manifests/machineconfigdaemon/daemonset.yaml

```diff
@@ -64,15 +50,3 @@ spec:
         - name: rootfs
           hostPath:
             path: /
-        - name: var-run-dbus
-          hostPath:
-            path: /var/run/dbus
-        - name: run-systemd
-          hostPath:
-            path: /run/systemd
-        - name: etc-ssl-certs
-          hostPath:
-            path: /etc/ssl/certs
-        - name: etc-mcd
-          hostPath:
-            path: /etc/machine-config-daemon
```

Comment:
wait wat, I've removed all the above (and regenerated bindata) but the masters journal still have logs like:
```
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.149635    1455 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "rootfs" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-rootfs") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.149663    1455 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "etc-mcd" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-etc-mcd") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250259    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "etc-ssl-certs" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-etc-ssl-certs") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250321    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "rootfs" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-rootfs") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250352    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "etc-mcd" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-etc-mcd") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250378    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "var-run-dbus" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-var-run-dbus") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250401    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "run-systemd" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-run-systemd") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250429    1455 reconciler.go:252] operationExecutor.MountVolume started for volume "machine-config-daemon-token-m78t4" (UniqueName: "kubernetes.io/secret/28548d72-6f77-11e9-9ce6-127a8ecd7226-machine-config-daemon-token-m78t4") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250502    1455 operation_generator.go:571] MountVolume.SetUp succeeded for volume "etc-mcd" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-etc-mcd") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250526    1455 operation_generator.go:571] MountVolume.SetUp succeeded for volume "run-systemd" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-run-systemd") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250529    1455 operation_generator.go:571] MountVolume.SetUp succeeded for volume "etc-ssl-certs" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-etc-ssl-certs") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250558    1455 secret.go:182] Setting up volume machine-config-daemon-token-m78t4 for pod 28548d72-6f77-11e9-9ce6-127a8ecd7226 at /var/lib/kubelet/pods/28548d72-6f77-11e9-9ce6-127a8ecd7226/volumes/kubernetes.io~secret/machine-config-daemon-token-m78t4
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250566    1455 operation_generator.go:571] MountVolume.SetUp succeeded for volume "rootfs" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-rootfs") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
May 05 20:48:38 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:38.250594    1455 operation_generator.go:571] MountVolume.SetUp succeeded for volume "var-run-dbus" (UniqueName: "kubernetes.io/host-path/28548d72-6f77-11e9-9ce6-127a8ecd7226-var-run-dbus") pod "machine-config-daemon-224jf" (UID: "28548d72-6f77-11e9-9ce6-127a8ecd7226")
```
what does that even mean? how is the kubelet coming up with those mounts if I'm not specifying them anymore?! Triple checked that it's picking up my code. The job is here https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/651/pull-ci-openshift-machine-config-operator-master-e2e-aws-upgrade/206/artifacts/e2e-aws-upgrade/nodes/

cc @derekwaynecarr if you have any pointer

Reply:
also, on failure, I can see the MCD running but kube reporting volumes ok only after it's running? shouldn't the container wait to run if not all mounts are there?

```
May 05 20:48:41 ip-10-0-159-115 root[14807]: machine-config-daemon[14756]: error loading pending config open /etc/machine-config-daemon/state.json: no such file or directory
May 05 20:48:41 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:41.391681    1455 kubelet.go:1960] SyncLoop (PLEG): "machine-config-daemon-224jf_openshift-machine-config-operator(28548d72-6f77-11e9-9ce6-127a8ecd7226)", event: &pleg.PodLifecycleEvent{ID:"28548d72-6f77-11e9-9ce6-127a8ecd7226", Type:"ContainerStarted", Data:"f52e0ef624d257ca5e57bffeca1d0f2f3c2d53376a8ad8ec7a57f96eac8b5bfb"}
May 05 20:48:41 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:41.391752    1455 kubelet_pods.go:1329] Generating status for "machine-config-daemon-224jf_openshift-machine-config-operator(28548d72-6f77-11e9-9ce6-127a8ecd7226)"
May 05 20:48:41 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:41.392121    1455 volume_manager.go:350] Waiting for volumes to attach and mount for pod "machine-config-daemon-224jf_openshift-machine-config-operator(28548d72-6f77-11e9-9ce6-127a8ecd7226)"
May 05 20:48:41 ip-10-0-159-115 hyperkube[1455]: I0505 20:48:41.392215    1455 volume_manager.go:383] All volumes are attached and mounted for pod "machine-config-daemon-224jf_openshift-machine-config-operator(28548d72-6f77-11e9-9ce6-127a8ecd7226)"
```

Reply:
cc @sjenning and @bchilds as I am out tomorrow.